### PR TITLE
use authorize edition to edit node

### PR DESCRIPTION
**Pull request by nicolasThal**

### CODE REVIEW

File: Backoffice/AuthorizeEdition/AuthorizeEditionInterface.php [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace OpenOrchestra\Backoffice\AuthorizeEdition;
+
```
Comment (itkg-alavieille) [inline Backoffice/AuthorizeEdition/AuthorizeEditionInterface.php]: remove one empty line

File: BackofficeBundle/DependencyInjection/OpenOrchestraBackofficeExtension.php [hunk 2]
```diff
@@ -65,6 +65,7 @@ public function load(array $configs, ContainerBuilder $container)
         $loader->load('validator.yml');
         $loader->load('initializer.yml');
         $loader->load('authorize_status_change.yml');
+        $loader->load('authorizeedition.yml');
```
Comment (itkg-alavieille) [inline BackofficeBundle/DependencyInjection/OpenOrchestraBackofficeExtension.php]: may be rename file by authorize_edition like others

File: BackofficeBundle/Resources/config/authorizeedition.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    open_orchestra_backoffice.authorize_edition.statusable:
+        class: %open_orchestra_backoffice.authorize_edition.statusable.class%
+        tags:
+            - { name: open_orchestra_backoffice.authorize_edition.strategy }
+    open_orchestra_backoffice.authorize_edition.node_version:
+        class: %open_orchestra_backoffice.authorize_edition.node_version.class%
+        arguments:
+            - @open_orchestra_model.repository.node
+        tags:
+            - { name: open_orchestra_backoffice.authorize_edition.strategy }
```
Comment (itkg-alavieille) [inline BackofficeBundle/Resources/config/authorizeedition.yml]: eof

File: BackofficeBundle/DependencyInjection/OpenOrchestraBackofficeExtension.php (hunk 2 above)
Comment (nicolasThal) [inline BackofficeBundle/DependencyInjection/OpenOrchestraBackofficeExtension.php]: no, yml file should not have _ in their name

Comment (itkg-alavieille) [inline BackofficeBundle/DependencyInjection/OpenOrchestraBackofficeExtension.php]: ok so can you rename ``authorize_status_change.yml`` in your PR

**Comment by nicolasThal** [comment]
@itkg-alavieille pr updated

**Comment by nicolasThal** [comment]
I have added the modification to not use the stored proc.

I still have to pass the current version (transformer update).

Are everyone ok with the modification for now?

File: BackofficeBundle/Manager/NodeManager.php [hunk 4]
```diff
@@ -69,22 +69,29 @@ public function __construct(
...
-        $node = $this->nodeRepository->findOneByNodeIdAndLanguageAndSiteIdAndVersion($nodeId, $language, $siteId);
+        $node = $this->nodeRepository->findOneByNodeIdAndLanguageAndSiteIdAndVersion($nodeId, $language, $siteId, $version);
+        $lastNode = $this->nodeRepository->findOneByNodeIdAndLanguageAndSiteIdInLastVersion($nodeId, $language, $siteId);
+        $lastNodeVersion = $lastNode->getVersion();
         $status = $this->getEditableStatus($node);
         if ($status === null) {
             $status = $this->statusRepository->findOneByInitial();
         }
-        $newNode = $this->nodeManager->duplicateNode($nodeId, $siteId, $language, $status->getId());
+        $newNode = clone $node;
```
Comment (itkg-alavieille) [inline BackofficeBundle/Manager/NodeManager.php]: i'm not sure but if you clone the node, you should be use

Comment (nicolasThal) [inline BackofficeBundle/Manager/NodeManager.php]: Did you wrote everything you wanted ?

Comment (itkg-alavieille) [inline BackofficeBundle/Manager/NodeManager.php]: yes

**Pull request merged**